sound/soc/sof/pm.c

```diff
@@ -423,3 +423,86 @@ int snd_sof_suspend(struct device *dev)
 	return sof_suspend(dev, false);
 }
 EXPORT_SYMBOL(snd_sof_suspend);
+
+static int snd_sof_d0ix_update(struct snd_sof_dev *sdev, bool d0i3_enable)
+{
+	int ret;
+
+	if (sdev->dsp_state == SOF_DSP_D0I3 && d0i3_enable) {
```

Comment:
I think it's really better to query the pm_runtime state (active or suspended) and only deal with D0ix when the pm_runtime state is 'active'.
You are duplicating work and information here.

Reply:
Thanks, let me change this and get back.